src/plugin.js

```diff
@@ -1,20 +1,70 @@
-import _ from 'lodash';
+import Handlebars from 'handlebars';
 
 const plugin = (editor) => {
   editor.addButton('reagent', {
-    text: 'reagent',
-    icon: false,
-    onclick: () => {
-      // Open window
-      editor.windowManager.open({
-        title: 'reagent plugin',
-        body: [
-          { type: 'textbox', name: 'title' }
-        ],
+    tooltip: 'Reagent',
+    icon: 'reagent',
+
+    onclick() {
+      let window = editor.windowManager.open({
+        height: 112,
+        width: 460,
+        title: 'Insert reagent',
+
+        buttons: [{
+          text: 'Ok',
+          classes: 'widget btn primary',
+          onclick() {
+            window.submit();
+          }
+        }, {
+          text: 'Cancel',
+          onclick() {
+            window.close();
+          }
+        }],
+
+        body: [{
+          type: 'textbox',
+          name: 'reagent',
+          label: 'Select reagent'
+        }],
+
         onsubmit(e) {
-          // Insert content when the window form is submitted
-          const kebabbyString = _.kebabCase(e.data.title);
-          editor.insertContent(kebabbyString);
+          let value = e.data.reagent;
+          editor.insertContent(`<span class="widget reagent">${value}</span>&nbsp;`);
+        }
+      });
+
+      let modal = $('#' + window._id);
+      let input = $(modal.find('input.mce-textbox').get(0));
```

Comment:
Not sure about the syntax but both lines above could be written using a single CSS selector:

    let input = $('#' + window._id + ' input.mce-textbox:first-of-type');